vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ModuleLinkManager.java

```diff
@@ -104,4 +110,14 @@ protected void deleteLinkRepo(IModuleLink link) {
         moduleLinkRepo.delete((ModuleLink) link);
     }
 
+    @Override
+    public HashSet <ISpace> findModuleLinksFromModuleId(String moduleId) {
+        List <ModuleLink> moduleLinks = moduleLinkRepo.getModuleLinks(moduleId);
+        HashSet <ISpace> spaces = new HashSet <>();
+        for (ModuleLink link: moduleLinks) {
+            spaces.add(link.getSpace());
+        }
+        return spaces;
```

Comment:
the above can be a one-liner using the stream api

Reply:
fixed

Reply:
116-118 can be reduced to one line (using `collect`).

Reply:
used (collect). to reduce lines